core/quorum_validation.go

```diff
@@ -2,15 +2,26 @@ package core
 
 import (
 	"bytes"
+	"fmt"
+	"sync"
 
 	ipfsnode "github.com/ipfs/go-ipfs-api"
 	"github.com/rubixchain/rubixgoplatform/block"
 	"github.com/rubixchain/rubixgoplatform/contract"
+	"github.com/rubixchain/rubixgoplatform/core/wallet"
 	"github.com/rubixchain/rubixgoplatform/did"
 	"github.com/rubixchain/rubixgoplatform/token"
 	"github.com/rubixchain/rubixgoplatform/util"
 )
 
+type TokenStateCheckResult struct {
+	Token                 string
+	Status                bool
```

Comment:
shall we change this bool to enum/const of known statuses?

Reply:
Read 1 & 2,
3. If the only goal ever with TokenStateCheckResult is whether it is exhausted or not,
It will be more clear if we just have a key called Exhausted(bool) instead of a Status (bool)

Reply:
I would still prefer to go with enum for future development compatibility.